src/audio_core/audio_renderer.cpp

```diff
@@ -315,10 +327,24 @@ void AudioRenderer::QueueMixedBuffer(Buffer::Tag tag) {
 }
 
 void AudioRenderer::ReleaseAndQueueBuffers() {
-    const auto released_buffers{audio_out->GetTagsAndReleaseBuffers(stream)};
-    for (const auto& tag : released_buffers) {
-        QueueMixedBuffer(tag);
+    if (!stream->IsPlaying()) {
+        return;
+    }
+
+    {
+        std::scoped_lock l{lock};
+        const auto released_buffers{audio_out->GetTagsAndReleaseBuffers(stream)};
+        for (const auto& tag : released_buffers) {
+            QueueMixedBuffer(tag);
+        }
     }
+
+    const f32 sampleRate = static_cast<f32>(GetSampleRate());
```

Comment:
```suggestion
    const f32 sample_rate = static_cast<f32>(GetSampleRate());
```

Ditto for other variables below